crate_universe/src/context/crate_context.rs

```diff
@@ -25,6 +25,10 @@ pub struct CrateDependency {
     /// Some dependencies are assigned aliases. This is tracked here
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub alias: Option<String>,
+
+    /// Where to acquire the source of this dependency.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub(crate) source_annotation: Option<SourceAnnotation>,
```

Comment:
@jayzhuang This appears to have exploded the size of the lockfiles. Is it not sufficient to use the `id` here to parse this info from the larger context object? 

https://github.com/bazelbuild/rules_rust/blob/c2d74a23835dd7f03d35f17848c21979f7dfd778/crate_universe/src/context/crate_context.rs#L303-L307

Reply:
I got #3334 to address this, PTAL. I can't get this information from the larger context, but I can only plumb local paths around. This drops serialized http and git source annotations, which contributes to the dominating majority of lockfile size bloats.